src/dom/DomEvent.js

```diff
@@ -224,9 +224,9 @@ export function getMousePosition(e, container) {
 
 	var scaleX = rect.width / container.offsetWidth || 1;
 	var scaleY = rect.height / container.offsetHeight || 1;
-	return new Point(
-		e.clientX / scaleX - rect.left - container.clientLeft,
-		e.clientY / scaleY - rect.top - container.clientTop);
+	var localX = (e.clientX - rect.left) / scaleX;
+	var localY = (e.clientY - rect.top) / scaleY;
+	return abs ? new Point(localX + rect.left, localY + rect.top) : new Point(localX, localY);

```

Comment:
is it possible to leave `getMousePosition` less complex? E.g. move scaled point calculation to a separate util function, which will be used by `getMousePosition` *and* inside `Draggable`?

Reply:
i think that shall make it more complex. Any calling of function has some overhead. Ant that's mouse event. Any single function call shall make slower response to user. 

Now getMouseEvent looks as only method to get right coordinates of mouse event. It's used now everywhere at Leaflet. I think that's right behavior. 

Third argument i've added is not simple for understanding and even for description... but it's needed in that situation I set it to true. We can try to avoid it... but i think it's better for now to add it. Ternary operator is very atomic and fast.

Reply:
> Third argument i've added is not simple for understanding and even for description

That's what I meant under "complexity".

Right now `getMousePosition` is easy to **reason about**. With `abs` argument it's much harder to do it.
Same with `mapContainer` option in `Draggable`. Right now `Draggable` doesn't care about map container at all - and in most cases it shouldn't.

Personally, I would rather not add this complexity to Leaflet core, unless there's simpler / more readable solution for that.
It's a corner case and 99% of users do not need this. So I'd stick with current, less complex solution.

From the other hand, I understand that you need this to work and I really appreciate your efforts!
But maybe it's possible to pull it out into a plugin (maybe with some additional work on Leaflet side to make it possible)? If I can help you with that - just let me know :)

That being said, it's just my personal opinion, and maybe other core contributors have other thoughts.
cc @mourner @perliedman @IvanSanchez @ghybs @yohanboniface

Reply:
I agree this makes a core function a bit more complex, but pulling this fix out of the core to a plugin sounds a bit weird to me. 

Would it be possible to name and document the `abs` better?

Reply:
Hi,

It seems to me here that the `abs` parameter is unnecessary: the only 2 places where this PR uses it are in `Draggable`'s `_onDown` and `_onMove` listeners. The results are then subtracted to compute the move offset (extent).

So we have: (pseudo-code)

```javascript
offset = movePointAbs - startPointAbs
       = getMousePosition(movePoint, mapContainer, true) - getMousePosition(startPoint, mapContainer, true)
       = Point(move_localX + move_rect.left, move_localY + move_rect.top) - Point(start_localX + start_rect.left, start_localY + rect.top)
       = Point(move_localX - start_localX, move_localY - start_localY) + Point(move_rect.left - start_rect.left,  move_rect.top - start_rect.top)
```

Since it is an _extremely_ corner case that `mapContainer` changes position during the drag operation, then `move_rect === start_rect` and we have:
```javascript
Point(move_rect.left - start_rect.left,  move_rect.top - start_rect.top) === 0
offset = Point(move_localX - start_localX, move_localY - start_localY)
       = Point(move_localX, move_localY) - Point(start_localX, start_localY)
       = getMousePosition(movePoint, mapContainer) - getMousePosition(startPoint, mapContainer)
       = movePoint - startPoint
```

Furthermore, the description for `getMousePosition` sounds pretty clear to me that it is relative to the container:

> Gets normalized mouse position from a DOM event relative to the `container`

The only case where including the start and in-movement offsets in the computation would be interesting is when we expect the map container to be shifted while dragging. In that case, the drag could indeed continue smoothly, even though the mouse / finger no longer holds the same spot… But I think we could expect the user to already feel a discrepancy (when the map container moves), and wonder why he/she is no longer holding the same spot.

Without including this `abs` offset, on the contrary the map "jumps" back to where the mouse / finger was holding, which IMHO makes the user experience less disruptive.
Demo: https://plnkr.co/edit/poTcoZRUYKBRxvWnHoZp?p=preview